[sonic-snmpagent] SONiC physical entity mib extension

Junchao-Mellanox opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

Add new physical entity fan drawer, fan, fan tachometers, PSU, PSU sensor, chassis sensors to physical entity mib.

Implement all mib objects defined in EntPhysicalEntry of RFC 2737.

**- How I did it**

Refactor rfc2737.py to have different mib updater for different physical entity: XcvrCacheUpdater, FanCacheUpdater, FanDrawerCacheUpdater, PsuCacheUpdater, ThermalCacheUpdater.

**- How to verify it**

Manual test on MSN2410

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

Junchao-Mellanox committed:
SONiC MIB Extension feature first commit

Junchao-Mellanox committed:
Add unit test cases and fix found issues

Junchao-Mellanox committed:
Add more comments

Junchao-Mellanox committed:
fix issue found in manual tests

Junchao-Mellanox committed:
Fix issue found in manual test

Junchao-Mellanox committed:
Fix issue: PSU fan is not in entity mib

Junchao-Mellanox committed:
Add chassis name and description

keboliu commented on src/sonic_ax_impl/mibs/__init__.py:
```diff
@@ -43,6 +54,13 @@
     "tx4power":     43,
 }
 
+PSU_SENSOR_PART_ID_MAP = {
```
Is this map common for all vendors?

Junchao-Mellanox replied:
Yes, all sensor values can get from platform API. If vendor doesn't implement the API, we will give it a None value in psud and it won't be displayed in mib.

jleveque committed:
[setup.py] Add 'wheel' to 'setup_requires' (#160)

Add 'wheel' to the list of packages required for building the package. This way it will be implicitly installed at build time, preventing the need to intstall the 'wheel' package explicitly in our build environment.

Junchao-Mellanox committed:
Change physical index generating rule

SuvarnaMeenakshi committed:
Implementation changes for CiscoBgp4MIB (#158)

* [BGP]:  Currently the data required for CiscoBgp4MIB is retrieved from bgpd deamon. snmp_ax_impl connects to bgpd daemon via tcp socket and retreives the BGP neighbor information required for CiscoBgp4MIB. This design is modified to support multi-asic platform. The data required by CiscoBgp4MIB can be populated in STATE_DB by a new daemon in BGP docker. Changes made:
- snmp_ax_impl to retrieve NEIGH_STATE_TABLE from STATE_DB. This change will affect both single and multi asic platforms. 
- Update bgp MIB unit tests to use STATE_DB data instead of using vtysh socket.

SuvarnaMeenakshi committed:
[ciscoSwitchQosMIB]: Remove invocation of update_data function during (#161)

reinit_data. udpate_data and reinit_data will be invoked by periodic
events. update_data need not be invoked by reinit_data again.
Additional invocation of update_data can cause increase in running time
on multi-asic platform causing Agentx socket connection to break.

Signed-off-by: SuvarnaMeenakshi <sumeenak@microsoft.com>

qiluo-msft committed:
Fix: correctly handle that lldp_loc_man_addr contains only IPv6 address without IPv4 address (#164)

* Fix mgmt_ip_sub_oid default value to prevent later exception
* Refix
* Reimplement LLDPLocManAddrUpdater
* Fix test case
* Revert some back and fix lookup()

SuvarnaMeenakshi committed:
[LLDP]: Update init_db to load global database config (#166)

[LLDP]: Update init_db to load global database config
before initializing SonicV2Connector class.
LLDPLocManAddrUpdater requires information from host database hence uses init_db() to initialize the host db.
For unit-testing multi-asic code path, there are mock db files provided for global_db and namespace specific dbs.
As init_db did not load global database config, unit-test mock db was not loading global_db database files for LLDPLocManAddrUpdater unit-testing. Fixed this by loading global database config in init_db function. Fixed by:
Loading global database config in init_db()
unit-test: Loading mock db files from "global_db" director for multi-asic platform if namespace is empty or None.

qiluo-msft committed:
Add a test case for LLDP_LOC_CHASSIS.lldp_loc_man_addr has only one IPv6 address (#167)

* Add a test case for LLDP_LOC_CHASSIS.lldp_loc_man_addr has only one IPv6 address
* Remove IPv4 on asicN namespace's appl_db

Junchao-Mellanox committed:
Merge branch 'master' into phy-mibs

Junchao-Mellanox commented:
https://github.com/Azure/sonic-snmpagent/pull/168
